### ISSUE TITLE
Tabix I/O does not work

### ISSUE THREAD

**Issue by holtgrewe** [seqan/seqan#1835]
```
./bin/demo_tabix_io_tabix_vcf minimal.vcf.gz 20 14300 14375
Contig 20 not found!
```

```
$ tabix minimal.vcf.gz 20:14300-14375
20	14370	rs6054257	G	A	29	PASS	NS=3;DP=14;AF=0.5;DB;H2	GT:GQ:DP:HQ	0|0:48:1:51,51	1|0:48:8:51,51	1/1:43:5:.,.
```

[vcf.zip](https://github.com/seqan/seqan/files/434858/vcf.zip)

**Comment by holtgrewe** [comment]
@weese :beetle:

**Comment by holtgrewe** [comment]
;)

**Comment by holtgrewe** [comment]
OK, I figured it out. The problem is that the `'\0'` characters in the contig names from the TBI file are copied over into the name store of the index. Thus, the names are `"20\0"` which when using `operator==()` is not equal to `"20"`.

:scream:

**Comment by rrahn** [comment]
@holtgrewe Can you fix it?

**Comment by holtgrewe** [comment]
@rrahn OK, just added a patch for 3 more lines to #1837. The fix is to trim the trailing `'\0'`s so they don't end up polluting the strings.

**Comment by rrahn** [comment]
Fixed by #1837
